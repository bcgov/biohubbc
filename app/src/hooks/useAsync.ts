import { useRef } from 'react';

export type AsyncFunction<Q extends any[], R> = (...args: Q) => Promise<R>;

export type WrappedAsyncFunction<Q extends any[], S> = (...args: Q) => Promise<S>;

/**
 * Wraps an async function to prevent duplicate calls if the previous call is still pending.
 *
 * Example:
 *
 * ```
 * const myAsyncFunction = useAsync(
 *   (param: number) => asyncFunction(param1)
 * )
 *
 * await myAsyncFunction(1) // calls `asyncFunction`, where param=1
 * (call 1 is pending)
 * await myAsyncFunction(2) // returns promise from call 1
 * (call 1 is fulfilled)
 * await myAsyncFunction(2) // calls `asyncFunction`, where param=2
 * (call 2 is pending)
 * ```
 *
 * @template Q
 * @template R
 * @param {AsyncFunction<Q, R>} asyncFunction the async function to wrap
 * @param {IUseAsyncOptions} [options]
 * @return {*}  {WrappedAsyncFunction<Q, R>}
 */
export const useAsync = <Q extends any[], R>(asyncFunction: AsyncFunction<Q, R>): WrappedAsyncFunction<Q, R> => {
  const ref = useRef<Promise<R>>();

  const isPending = useRef(false);

  const wrappedAsyncFunction: WrappedAsyncFunction<Q, R> = async (...args) => {
    if (ref.current && isPending.current) {
      return ref.current;
    }

    isPending.current = true;

    ref.current = asyncFunction(...args)
      .then(
        (response: R) => {
          isPending.current = false;

          return response;
        },
        (error) => {
          isPending.current = false;

          throw error;
        }
      )
      .catch((error) => {
        isPending.current = false;

        throw error;
      });

    return ref.current;
  };

  return wrappedAsyncFunction;
};
