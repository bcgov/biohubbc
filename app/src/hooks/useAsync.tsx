import { useRef } from 'react';

export type AsyncFunction<AFArgs extends any[], AFResponse> = (...args: AFArgs) => Promise<AFResponse>;

/**
 * Wraps an async function to prevent duplicate calls if the previous call is still pending.
 *
 * Example:
 *
 * ```
 * const myAsyncFunction = useAsync(
 *   (param: number) => asyncFunction(param1)
 * )
 *
 * await myAsyncFunction(1) // calls `asyncFunction`, where param=1
 * // call 1 is pending
 * await myAsyncFunction(2) // returns pending promise from call 1
 * await myAsyncFunction(2) // returns pending promise from call 1
 * // call 1 is fulfilled
 * await myAsyncFunction(2) // calls `asyncFunction`, where param=2
 * // call 2 is pending
 * ```
 *
 * @template AFArgs `AsyncFunction` argument types.
 * @template AFResponse `AsyncFunction` response type.
 * @param {AsyncFunction<AFArgs, AFResponse>} asyncFunction the async function to wrap
 * @return {*}  {AsyncFunction<AFArgs, AFResponse>}
 */
export const useAsync = <AFArgs extends any[], AFResponse>(
  asyncFunction: AsyncFunction<AFArgs, AFResponse>
): AsyncFunction<AFArgs, AFResponse> => {
  const ref = useRef<Promise<AFResponse>>();

  const isPending = useRef(false);

  const wrappedAsyncFunction: AsyncFunction<AFArgs, AFResponse> = async (...args) => {
    if (ref.current && isPending.current) {
      return ref.current;
    }

    isPending.current = true;

    ref.current = asyncFunction(...args).then(
      (response: AFResponse) => {
        isPending.current = false;

        return response;
      },
      (error) => {
        isPending.current = false;

        throw error;
      }
    );

    return ref.current;
  };

  return wrappedAsyncFunction;
};
