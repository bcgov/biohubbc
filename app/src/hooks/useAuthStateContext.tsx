import { useContext } from 'react';
import { AuthStateContext, IAuthState } from '../contexts/authStateContext';

/**
 * Returns an instance of `IAuthState` from `AuthStateContext`.
 *
 * @return {*}  {IAuthState}
 */
export const useAuthStateContext = (): IAuthState => {
  const context = useContext(AuthStateContext);

  if (!context) {
    throw Error(
      'AuthStateContext is undefined, please verify you are calling useAuthStateContext() as child of an <AuthStateProvider> component.'
    );
  }

  return context;
};
