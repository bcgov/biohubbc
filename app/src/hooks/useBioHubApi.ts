import { useKeycloak } from '@react-keycloak/web';
import axios from 'axios';
import { IProject, IProjectPostObject } from 'interfaces/project-interfaces';
import { IActivity, ICreateActivity, ICreateProjectResponse, ITemplate } from 'interfaces/useBioHubApi-interfaces';
import moment from 'moment';
import { useMemo } from 'react';

const API_HOST = process.env.REACT_APP_API_HOST;
const API_PORT = process.env.REACT_APP_API_PORT;

const API_URL = (API_PORT && `${API_HOST}:${API_PORT}`) || API_HOST || 'api-dev-biohubbc.apps.silver.devops.gov.bc.ca';

/**
 * Returns an instance of axios with baseURL and authorization headers set.
 *
 * @return {*}
 */
const useApi = () => {
  const { keycloak } = useKeycloak();
  const instance = useMemo(() => {
    return axios.create({
      headers: {
        Authorization: `Bearer ${keycloak?.token}`
      },
      baseURL: API_URL
    });
  }, [keycloak]);

  return instance;
};

/**
 * Returns a set of supported api methods.
 *
 * @return {object} object whose properties are supported api methods.
 */
export const useBiohubApi = () => {
  const api = useApi();

  /**
   * Get all projects.
   *
   * @return {*}  {Promise<IProject[]>}
   */
  const getProjects = async (): Promise<IProject[]> => {
    // const { data } = await api.get(`/api/projects`);

    // return data;

    // TODO: stub for development
    return [
      {
        id: 1,
        name: 'Project Name 1',
        objectives: 'Project Objectives 1',
        scientific_collection_permit_number: '123456',
        management_recovery_action: 'A',
        location_description: 'Location Description 1',
        start_date: moment().toISOString(),
        end_date: moment().toISOString(),
        results: 'Results 1',
        caveats: 'Caveats 1',
        comments: 'Comments 1'
      },
      {
        id: 2,
        name: 'Project Name 2',
        objectives: 'Project Objectives 2',
        scientific_collection_permit_number: '123456',
        management_recovery_action: 'A',
        location_description: 'Location Description 2',
        start_date: moment().toISOString(),
        end_date: moment().toISOString(),
        results: 'Results 2',
        caveats: 'Caveats 2',
        comments: 'Comments 2'
      },
      {
        id: 3,
        name: 'Project Name 3',
        objectives: 'Project Objectives 3',
        scientific_collection_permit_number: '123456',
        management_recovery_action: 'A',
        location_description: 'Location Description 3',
        start_date: moment().toISOString(),
        end_date: moment().toISOString(),
        results: 'Results 3',
        caveats: 'Caveats 3',
        comments: 'Comments 3'
      },
      {
        id: 4,
        name: 'Project Name 4',
        objectives: 'Project Objectives 4',
        scientific_collection_permit_number: '123456',
        management_recovery_action: 'A',
        location_description: 'Location Description 4',
        start_date: moment().toISOString(),
        end_date: moment().toISOString(),
        results: 'Results 4',
        caveats: 'Caveats 4',
        comments: 'Comments 4'
      }
    ];
  };

  /**
   * Get a project based on its ID.
   *
   * @param {projectId} projectId
   * @return {*}  {Promise<IProject>}
   */
  const getProject = async (projectId: number): Promise<IProject> => {
    const { data } = await api.get(`/api/project/${projectId}`);

    return data;
  };

  /**
   * Create a new project.
   *
   * @param {IProjectPostObject} project
   * @return {*}  {Promise<ICreateProjectResponse>}
   */
  const createProject = async (project: IProjectPostObject): Promise<ICreateProjectResponse> => {
    const { data } = await api.post('/api/project', project);

    return data;
  };

  /**
   * Get a template based on its ID.
   *
   * @param {templateId} templateId
   * @return {*}  {Promise<ITemplate>}
   */
  const getTemplate = async (templateId: number): Promise<ITemplate> => {
    const { data } = await api.get(`/api/template/${templateId}`);

    return data;
  };

  /**
   * Create a new activity record.
   *
   * @param {ICreateActivity} activity
   * @return {*}  {Promise<IActivity>}
   */
  const createActivity = async (activity: ICreateActivity): Promise<IActivity> => {
    const { data } = await api.post('/api/activity', activity);

    return data;
  };

  /**
   * Fetch all code sets.
   *
   * @return {*}  {Promise<any>}
   */
  const getAllCodes = async (): Promise<any> => {
    const { data } = await api.get('/api/codes/');

    return data;
  };

  /**
   * Fetch the api json-schema spec.
   *
   * @return {*}  {Promise<any>}
   */
  const getApiSpec = async (): Promise<any> => {
    const { data } = await api.get('/api/api-docs/');

    return data;
  };

  return {
    getProjects,
    getProject,
    getTemplate,
    createProject,
    createActivity,
    getAllCodes,
    getApiSpec
  };
};
