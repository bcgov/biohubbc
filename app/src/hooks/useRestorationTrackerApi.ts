import axios from 'axios';
import useAdminApi from './api/useAdminApi';
import useAxios from './api/useAxios';
import useCodesApi from './api/useCodesApi';
import useDraftApi from './api/useDraftApi';
import useExternalApi from './api/useExternalApi';
import useProjectApi, { usePublicProjectApi } from './api/useProjectApi';
import useSearchApi, { usePublicSearchApi } from './api/useSearchApi';
import useUserApi from './api/useUserApi';
import usePermitApi from './api/usePermitApi';
import { useContext } from 'react';
import { ConfigContext } from 'contexts/configContext';

/**
 * Returns a set of supported api methods.
 *
 * @return {*} object whose properties are supported api methods.
 */
export const useRestorationTrackerApi = () => {
  const config = useContext(ConfigContext);
  const apiAxios = useAxios(config?.API_HOST);

  const project = useProjectApi(apiAxios);

  const permit = usePermitApi(apiAxios);

  const search = useSearchApi(apiAxios);

  const codes = useCodesApi(apiAxios);

  const draft = useDraftApi(apiAxios);

  const user = useUserApi(apiAxios);

  const admin = useAdminApi(apiAxios);

  const external = useExternalApi(axios);

  const publicApis = {
    project: usePublicProjectApi(apiAxios),
    search: usePublicSearchApi(apiAxios)
  };

  return {
    project,
    permit,
    search,
    codes,
    draft,
    user,
    admin,
    external,
    public: publicApis
  };
};
