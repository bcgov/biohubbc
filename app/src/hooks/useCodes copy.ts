import { IGetAllCodeSetsResponse } from 'interfaces/useCodesApi.interface';
import { useBiohubApi } from './useBioHubApi';
import useDataLoader, { DataLoader } from './useDataLoader';

/**
 * Hook that fetches app code sets.
 *
 * @export
 * @return {*}  {DataLoader<IGetAllCodeSetsResponse>}
 */
export default function useCodes(): DataLoader<IGetAllCodeSetsResponse> {
  const api = useBiohubApi();

  return useDataLoader<IGetAllCodeSetsResponse>(api.codes.getAllCodeSets);
}
