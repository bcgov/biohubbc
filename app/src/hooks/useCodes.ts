import { IGetAllCodeSetsResponse } from 'interfaces/useCodesApi.interface';
import { useEffect, useState } from 'react';
import { useBiohubApi } from './useBioHubApi';

export interface IUseCodes {
  codes: IGetAllCodeSetsResponse | undefined;
  isLoading: boolean;
  isReady: boolean;
}

/**
 * Hook that fetches app code sets.
 *
 * @export
 * @return {*}  {IUseCodes}
 */
export default function useCodes(): IUseCodes {
  const api = useBiohubApi();

  const [codes, setCodes] = useState<IGetAllCodeSetsResponse>();
  const [isLoading, setIsLoading] = useState<boolean>(false);
  const [isReady, setIsReady] = useState<boolean>(false);

  useEffect(() => {
    const fetchCodes = async () => {
      setIsLoading(true);

      const response = await api.codes.getAllCodeSets();

      setCodes(response);

      setIsReady(true);
    };

    if (codes) {
      return;
    }

    fetchCodes();
  }, [api.codes, codes]);

  return { codes, isLoading, isReady };
}
