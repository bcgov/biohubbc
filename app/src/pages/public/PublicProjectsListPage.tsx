import Box from '@material-ui/core/Box';
import Container from '@material-ui/core/Container';
import Paper from '@material-ui/core/Paper';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import Link from '@material-ui/core/Link';
import TableContainer from '@material-ui/core/TableContainer';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import { DATE_FORMAT } from 'constants/dateTimeFormats';
import { AuthStateContext } from 'contexts/authStateContext';
import { useRestorationTrackerApi } from 'hooks/useRestorationTrackerApi';
import { IGetProjectsListResponse } from 'interfaces/useProjectApi.interface';
import React, { useContext, useEffect, useState } from 'react';
import { Redirect, useHistory } from 'react-router';
import { getFormattedDate } from 'utils/Utils';
import { ProjectStatusType } from 'constants/misc';
import clsx from 'clsx';
import Chip from '@material-ui/core/Chip';
import makeStyles from '@material-ui/core/styles/makeStyles';
import { Theme } from '@material-ui/core/styles/createMuiTheme';
import Typography from '@material-ui/core/Typography';

const useStyles = makeStyles((theme: Theme) => ({
  chip: {
    padding: '0px 8px',
    borderRadius: '4px',
    color: 'white'
  },
  chipActive: {
    backgroundColor: theme.palette.warning.main
  },
  chipCompleted: {
    backgroundColor: theme.palette.success.main
  }
}));

const PublicProjectsListPage = () => {
  const { keycloakWrapper } = useContext(AuthStateContext);
  const restorationTrackerApi = useRestorationTrackerApi();
  const classes = useStyles();
  const history = useHistory();

  const [isLoading, setIsLoading] = useState(true);
  const [projects, setProjects] = useState<IGetProjectsListResponse[]>([]);

  useEffect(() => {
    const getProjects = async () => {
      const projectsResponse = await restorationTrackerApi.public.project.getProjectsList();

      setProjects(() => {
        setIsLoading(false);
        return projectsResponse;
      });
    };

    if (isLoading) {
      getProjects();
    }
  }, [restorationTrackerApi, isLoading]);

  const getChipIcon = (status_name: string) => {
    let chipLabel;
    let chipStatusClass;

    if (ProjectStatusType.ACTIVE === status_name) {
      chipLabel = 'ACTIVE';
      chipStatusClass = classes.chipActive;
    } else if (ProjectStatusType.COMPLETED === status_name) {
      chipLabel = 'COMPLETED';
      chipStatusClass = classes.chipCompleted;
    }

    return <Chip size="small" className={clsx(classes.chip, chipStatusClass)} label={chipLabel} />;
  };

  const navigateToPublicProjectPage = (id: number) => {
    history.push(`/projects/${id}`);
  };

  const getProjectsTableData = () => {
    const hasProjects = projects?.length > 0;

    if (!hasProjects) {
      return (
        <Table>
          <TableHead>
            <TableRow>
              <TableCell>Name</TableCell>
              <TableCell>Type</TableCell>
              <TableCell>Permits</TableCell>
              <TableCell>Contact Agency</TableCell>
              <TableCell>Start Date</TableCell>
              <TableCell>End Date</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            <TableRow>
              <TableCell colSpan={6}>
                <Box display="flex" justifyContent="center">
                  No Results
                </Box>
              </TableCell>
            </TableRow>
          </TableBody>
        </Table>
      );
    } else {
      return (
        <TableContainer>
          <Table>
            <TableHead>
              <TableRow>
                <TableCell>Name</TableCell>
                <TableCell>Type</TableCell>
                <TableCell>Permits</TableCell>
                <TableCell>Contact Agency</TableCell>
                <TableCell>Start Date</TableCell>
                <TableCell>End Date</TableCell>
                <TableCell>Completion Status</TableCell>
              </TableRow>
            </TableHead>
            <TableBody data-testid="project-table">
              {projects?.map((row) => (
                <TableRow key={row.id}>
                  <TableCell component="th" scope="row">
                    <Link
                      data-testid={row.name}
                      underline="always"
                      component="button"
                      variant="body2"
                      onClick={() => navigateToPublicProjectPage(row.id)}>
                      {row.name}
                    </Link>
                  </TableCell>
                  <TableCell>{row.project_type}</TableCell>
                  <TableCell>{row.permits_list}</TableCell>
                  <TableCell>{row.coordinator_agency}</TableCell>
                  <TableCell>{getFormattedDate(DATE_FORMAT.ShortMediumDateFormat, row.start_date)}</TableCell>
                  <TableCell>{getFormattedDate(DATE_FORMAT.ShortMediumDateFormat, row.end_date)}</TableCell>
                  <TableCell>{getChipIcon(row.completion_status)}</TableCell>
                </TableRow>
              ))}
            </TableBody>
          </Table>
        </TableContainer>
      );
    }
  };

  if (keycloakWrapper?.keycloak?.authenticated) {
    // User has a role
    return <Redirect to={{ pathname: '/admin/projects' }} />;
  }

  return (
    <Box my={4}>
      <Container maxWidth="xl">
        <Box mb={5}>
          <Typography variant="h1">Projects</Typography>
        </Box>
        <Paper>{getProjectsTableData()}</Paper>
      </Container>
    </Box>
  );
};

export default PublicProjectsListPage;
