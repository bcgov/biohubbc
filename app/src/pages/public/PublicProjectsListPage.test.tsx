import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { cleanup, render, waitFor } from '@testing-library/react';
import PublicProjectsListPage from './PublicProjectsListPage';
import { useRestorationTrackerApi } from 'hooks/useRestorationTrackerApi';

jest.mock('../../hooks/useRestorationTrackerApi');
const mockUseRestorationTrackerApi = {
  public: {
    project: {
      getProjectsList: jest.fn()
    }
  }
};

const mockRestorationTrackerApi = ((useRestorationTrackerApi as unknown) as jest.Mock<
  typeof mockUseRestorationTrackerApi
>).mockReturnValue(mockUseRestorationTrackerApi);

describe('PublicProjectsListPage', () => {
  beforeEach(() => {
    mockRestorationTrackerApi().public.project.getProjectsList.mockClear();
  });

  afterEach(() => {
    cleanup();
  });

  test('renders with a proper list of projects when completed', async () => {
    mockRestorationTrackerApi().public.project.getProjectsList.mockResolvedValue([
      {
        id: 1,
        name: 'Project 1',
        start_date: null,
        end_date: null,
        coordinator_agency: 'contact agency',
        project_type: 'project type',
        permits_list: '1, 2, 3',
        completion_status: 'Completed'
      }
    ]);

    const { getByText, getByTestId } = render(
      <MemoryRouter>
        <PublicProjectsListPage />
      </MemoryRouter>
    );

    await waitFor(() => {
      expect(getByTestId('project-table')).toBeInTheDocument();
      expect(getByText('COMPLETED')).toBeInTheDocument();
    });
  });

  test('renders with a proper list of projects when active', async () => {
    mockRestorationTrackerApi().public.project.getProjectsList.mockResolvedValue([
      {
        id: 1,
        name: 'Project 1',
        start_date: null,
        end_date: null,
        coordinator_agency: 'contact agency',
        project_type: 'project type',
        permits_list: '1, 2, 3',
        completion_status: 'Active'
      }
    ]);

    const { getByText, getByTestId } = render(
      <MemoryRouter>
        <PublicProjectsListPage />
      </MemoryRouter>
    );

    await waitFor(() => {
      expect(getByTestId('project-table')).toBeInTheDocument();
      expect(getByText('ACTIVE')).toBeInTheDocument();
    });
  });
});
