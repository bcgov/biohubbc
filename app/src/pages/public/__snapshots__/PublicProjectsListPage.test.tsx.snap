// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublicProjectsListPage renders with a proper list of projects when published 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-4"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthXl"
    >
      <div
        class="MuiBox-root MuiBox-root-5"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1"
        >
          Projects
        </h1>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="MuiTableContainer-root"
        >
          <table
            class="MuiTable-root"
          >
            <thead
              class="MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  Name
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  Type
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  Permits
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  Coordinator Agency
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  Start Date
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  End Date
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  Completion Status
                </th>
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root"
              data-testid="project-table"
            >
              <tr
                class="MuiTableRow-root"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-body"
                  role="cell"
                  scope="row"
                >
                  <button
                    class="MuiTypography-root MuiLink-root MuiLink-underlineAlways MuiLink-button MuiTypography-body2 MuiTypography-colorPrimary"
                    data-testid="Project 1"
                  >
                    Project 1
                  </button>
                </th>
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                >
                  project type
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                >
                  1, 2, 3
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                >
                  coordinator agency
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                />
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                />
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                >
                  <div
                    class="MuiChip-root makeStyles-chip-1 MuiChip-sizeSmall"
                  >
                    <span
                      class="MuiChip-label MuiChip-labelSmall"
                    />
                  </div>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`PublicProjectsListPage renders with a proper list of projects when unpublished 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-10"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthXl"
    >
      <div
        class="MuiBox-root MuiBox-root-11"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1"
        >
          Projects
        </h1>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="MuiTableContainer-root"
        >
          <table
            class="MuiTable-root"
          >
            <thead
              class="MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  Name
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  Type
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  Permits
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  Coordinator Agency
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  Start Date
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  End Date
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  Completion Status
                </th>
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root"
              data-testid="project-table"
            >
              <tr
                class="MuiTableRow-root"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-body"
                  role="cell"
                  scope="row"
                >
                  <button
                    class="MuiTypography-root MuiLink-root MuiLink-underlineAlways MuiLink-button MuiTypography-body2 MuiTypography-colorPrimary"
                    data-testid="Project 1"
                  >
                    Project 1
                  </button>
                </th>
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                >
                  project type
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                >
                  1, 2, 3
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                >
                  coordinator agency
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                />
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                />
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                >
                  <div
                    class="MuiChip-root makeStyles-chip-7 MuiChip-sizeSmall"
                  >
                    <span
                      class="MuiChip-label MuiChip-labelSmall"
                    />
                  </div>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
